flax/linen/combinators.py

```diff
@@ -112,3 +115,90 @@ def __call__(self, *args, **kwargs):
       else:
         outputs = layer(outputs)
     return outputs
+
+def _assign_shared_scope(parent, child):
+  if child.scope is not None and child.scope.parent == parent.scope:
+    # Child is a true child, overwrite its scope
+    object.__setattr__(child, 'scope', parent.scope)
+  else:
+    # Child has its own independent scope, overwrite
+    # parent scope, so that we preserve the sharing
+    object.__setattr__(parent, 'scope', child.scope)
+
+
+class Transparent(Module, Generic[M]):
+  """A Module that shares its scope with an inner Module. This combinator is useful
+  when you want to wrap a Module and extend itsfunctionality without changing the
+  parameter structure.
```

Comment:
I would also highlight the difference between 

class DenseLora(Dense): 
   ...

vs

class DenseLora(Transparent[Dense]):
   ...

(The important distinction is that Transparent replaces inheritance with composition, that allows to swap out implementation of inner module with Dense implementation (as long as it has compatible interface), without having to create separate DenseLora for each implementation.